File: backend-go/internal/modules/system/entity/sys_dept.go
package entity

import (
	"drip-admin/backend-go/internal/common"
	"time"
)

type SysDept struct {
	ID           common.Int64String  `gorm:"column:id;primaryKey" json:"id"`
	ParentID     common.Int64String  `gorm:"column:parent_id" json:"parentId"`
	DeptName     string              `gorm:"column:dept_name" json:"deptName"`
	DeptCode     string              `gorm:"column:dept_code" json:"deptCode"`
	LeaderUserID *common.Int64String `gorm:"column:leader_user_id" json:"leaderUserId"`
	Sort         int                 `gorm:"column:sort" json:"sort"`
	Status       int                 `gorm:"column:status" json:"status"`
	Deleted      int                 `gorm:"column:deleted" json:"deleted"`
	CreatedAt    *time.Time          `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt    *time.Time          `gorm:"column:updated_at" json:"updatedAt"`
}

func (SysDept) TableName() string { return "sys_dept" }

File: backend-python/app/modules/system/entity/sys_dept.py
from sqlalchemy import BigInteger, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.modules.system.entity.base import Base, TimestampMixin


class SysDept(TimestampMixin, Base):
    __tablename__ = "sys_dept"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    parent_id: Mapped[int] = mapped_column(BigInteger)
    dept_name: Mapped[str] = mapped_column(String(64))
    dept_code: Mapped[str] = mapped_column(String(64))
    leader_user_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    sort: Mapped[int] = mapped_column(Integer)
    status: Mapped[int] = mapped_column(Integer)
    deleted: Mapped[int] = mapped_column(Integer)
